ROOTFS/usr/bin/waggle-registration

```diff
@@ -123,52 +144,68 @@ def generate_token_from_key_and_cert(key, cert):
 
 def scan_ssh_key(s):
     try:
-        return re.search(r'ssh-rsa.*', s).group()
+        return re.search(r"ssh-rsa.*", s).group()
     except AttributeError:
-        raise ValueError('No matching SSH key:\n{}.'.format(s))
+        raise ValueError("No matching SSH key:\n{}.".format(s))
 
 
 def request_waggleca_info():
-    logger.info('Requesting waggleca info from %s.', cert_server)
+    logger.info("Requesting waggleca info from %s.", cert_server)
 
-    return {
-        'cacert': scan_certificate(make_request('certca'))
-    }
+    return {"cacert": scan_certificate(make_request("certca"))}
 
 
 def request_node_info(node_id):
-    logger.info('Requesting node info from %s.', cert_server)
+    logger.info("Requesting node info from %s.", cert_server)
 
-    response = make_request('node?{}'.format(node_id))
+    response = make_request("node?{}".format(node_id))
 
-    if 'cert file not found' in response:
-        raise ValueError('Certificate not found for {}.'.format(node_id))
+    if "cert file not found" in response:
+        raise ValueError("Certificate not found for {}.".format(node_id))
 
     return {
-        'cert': scan_certificate(response),
-        'key': scan_private_key(response),
-        'ssh_port': scan_ssh_port(response),
-        'ssh_key': scan_ssh_key(response),
-        'token': scan_token_or_failover(response),
+        "cert": scan_certificate(response),
+        "key": scan_private_key(response),
+        "ssh_port": scan_ssh_port(response),
+        "ssh_key": scan_ssh_key(response),
+        "token": scan_token_or_failover(response),
     }
 
 
 def get_hostname():
-    r = subprocess.check_output('hostname').decode().strip()
-
-    if 'SD' in r:
-        return 'SD'
-    if 'MMC' in r:
-        return 'eMMC'
-    return ''
-
+    r = subprocess.check_output("hostname").decode().strip()
+
+    if "SD" in r:
+        return "SD"
+    if "MMC" in r:
+        return "eMMC"
+    return ""
+
+
+def generate_node_id(interface):
```

Comment:
This is nice that now which device we use is just part of the config. I think the main question worth revisiting is whether you want the backups interfaces or not or if it should just fail.

If you use backup interfaces as the node ID, then we need to make sure that is used for the lifetime of the device. Note that `/etc/waggle/node_id` holds the ID you registered with so we need to make sure there's no lingering calls out to explicitly get a MAC address. (Not that it would work outside the ODROID specific setup anyway.)

Reply:
thanks!

so i debated the backup interfaces for a little bit… but i thought it was really important for this service to not fail in the field.

if this service crashes or just exits for a node in the field it will never register with beehive/beekeeper and we will never be able to login and fix it

and you mentioned that /etc/waggle/node_id contains the node_id registered with.  i actually dont see how that happens.  and looking on wb01 (one of the dell blades) i dont see that file.  are you sure this file gets created?

Reply:
After talking with sean we discovered that the `/etc/waggle/node-id` file is in the v2.x version of the branch with beekeeper registration (https://github.com/waggle-sensor/beekeeper-registration/pull/2)